components/calendar/test/calendar.visual-diff.js

```diff
@@ -0,0 +1,302 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-calendar', () => {
+
+	const visualDiff = new VisualDiff('calendar', __dirname);
+
+	let browser, page;
+
+	const firstCalendarOfPage = '#contains-today-diff-selected';
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.setViewport({width: 400, height: 1600, deviceScaleFactor: 2});
+		await page.goto(`${visualDiff.getBaseUrl()}/components/calendar/test/calendar.visual-diff.html`, {waitUntil: ['networkidle0', 'load']});
+		await page.bringToFront();
+	});
+
+	after(() => browser.close());
+
+	it('has no selected-value when selected-value not set', async function() {
+		const rect = await visualDiff.getRect(page, '#no-selected');
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+	});
+
+	it('month with first row entirely current month days and last row containing next month days', async function() {
+		const rect = await visualDiff.getRect(page, '#dec-2019');
+		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+	});
+
+	describe('localization', () => {
+
+		after(async() => {
+			await page.evaluate(() => document.querySelector('html').setAttribute('lang', 'en'));
+		});
+
+		[
+			'ar',
+			'da',
+			'de',
+			'en',
+			'es',
+			'fr',
+			'ja',
+			'ko',
+			'nl',
+			'pt',
+			'sv',
+			'tr',
+			'zh',
+			'zh-tw'
+		].forEach((lang) => {
+			it(`works correctly in ${lang}`, async function() {
+				await page.evaluate(lang => document.querySelector('html').setAttribute('lang', lang), lang);
+				const rect = await visualDiff.getRect(page, firstCalendarOfPage);
+				await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+		});
+	});
+
+	describe('style', () => {
+		afterEach(async() => {
+			await page.reload();
```

Comment:
Reloading the page is generally the safest approach, but is probably also the slowest.  If the time of the tests really blows up, one option is to reset the component between tests, by reseting focus and removing such classes - I've done that elsewhere... it's not bullet proof and may trip people up if they make changes to the tests, but a compromise in effort to reduce testing time.